src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -89,6 +89,13 @@ private static void SplitIRAndState(ControlInfoJson.Item control, string topPare
                                 },
                             });
 
+                            // Handle the case where invariantScript value of the property is not same as the default script.
+                            var invariantScript = control.Rules.First(rule => rule.Property == arg.Name)?.InvariantScript;
+                            if (invariantScript != null && invariantScript != arg.ScopeVariableInfo.DefaultRule)
```

Comment:
> if (invariantScript != null && invariantScript != arg.ScopeVariableInfo.DefaultRule) [](http://example.com/codeflow?start=27&length=85)

Is this too broad? It seems like it will catch all the expressions and put everything into entropy?